node_modules/oae-core/editprofile/editprofile.html

```diff
@@ -1,70 +1,134 @@
+        </div>
     </div>
-    <div class="form-group">
-        <label for="editprofile-email" class="control-label"><h4>__MSG__EMAIL__</h4></label>
-        <input type="text" id="editprofile-email" class="form-control required email maxlength-short" name="editprofile-email" placeholder="__MSG__ENTER_AN_EMAIL_ADDRESS__" value="${oae.data.me.email|encodeForHTMLAttribute}"/>
+    <div class="modal-footer">
+        <button type="button" class="btn btn-link">__MSG__BACK_TO_PROFILE__</button>
+        {if isValidProfile}
+            <button type="button" class="btn btn-primary">__MSG__DONE__</button>
+        {/if}
+    </div>
+--></div>
+
+<div id="editprofile-panel-main-template"><!--
+    <div class="modal-header">
+        {if isValidProfile}
+            <button type="button" class="close" data-dismiss="modal" aria-label="__MSG__CLOSE__">
+                <span aria-hidden="true">&#215;</span>
+            </button>
+        {/if}
+        <h3 id="editprofile-modal-title">
+            {if isValidProfile}
+                __MSG__EDIT_DETAILS__
+            {else}
+                __MSG__PLEASE_ENTER_YOUR_DETAILS__
+            {/if}
+        </h3>
     </div>
-    <h4>__MSG__PROFILE_VISIBILITY__</h4>
-    {var tenant = oae.api.util.security().encodeForHTML(oae.data.me.tenant.displayName)}
-    <div class="row oae-large-options-container">
-        <div class="col-sm-4 text-center">
-            <label for="oae-visibility-private" class="{if oae.data.me.visibility === 'private'} checked{/if}">
-                <i class="fa fa-check hide"></i>
-                <div class="oae-large-options well">
-                    <input type="radio" id="oae-visibility-private" value="private" name="oae-visibility-group" class="pull-left" {if oae.data.me.visibility === 'private'} checked="checked"{/if} tabindex="0"/>
-                    <i class="fa fa-oae-private large text-center"></i>
-                    <span class="oae-threedots">__MSG__PRIVATE__</span>
+    <form id="editprofile-form" role="form">
+        <div id="editprofile-container" class="modal-body">
+            {if !isValidProfile}
+                <h4 id="editprofile-plea">__MSG__ADDING_YOUR_FULL_NAME__</h4>
+            {/if}
+            <div class="form-group">
+                <label for="editprofile-name" class="control-label"><h4>__MSG__NAME__</h4></label>
+                <input type="text" id="editprofile-name" class="form-control required maxlength-short displayname" name="editprofile-name" placeholder="__MSG__ENTER_A_NAME__" value="${profile.displayName|encodeForHTMLAttribute}"/>
+            </div>
+            <div class="form-group">
+                <label for="editprofile-email" class="control-label"><h4>__MSG__EMAIL__</h4></label>
+                <input type="text" id="editprofile-email" class="form-control required email maxlength-short" name="editprofile-email" placeholder="__MSG__ENTER_AN_EMAIL_ADDRESS__" value="${profile.email|encodeForHTMLAttribute}"/>
```

Comment:
There are a few odd things about the current workflow. For example, when creating a new account and specifying your email address, you are then taken to the "Edit profile" modal where your email address is blank again with a validate message underneath.

I think it would be more logical for the input box to contain the latest email address (unverified if there is one, verified otherwise). That also removes the need to add the email address to the validation message underneath.